ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -15,10 +15,57 @@
  */
 package com.google.android.ground.ui.datacollection
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.LiveDataReactiveStreams
+import com.google.android.ground.model.submission.Submission
+import com.google.android.ground.repository.SubmissionRepository
+import com.google.android.ground.rx.Loadable
+import com.google.android.ground.rx.annotations.Hot
 import com.google.android.ground.ui.common.AbstractViewModel
+import com.google.android.ground.ui.common.LocationOfInterestHelper
+import io.reactivex.Flowable
+import io.reactivex.processors.BehaviorProcessor
+import io.reactivex.processors.FlowableProcessor
 import javax.inject.Inject
 
 /**
  * View model for the Data Collection fragment.
  */
-class DataCollectionViewModel @Inject internal constructor() : AbstractViewModel()
\ No newline at end of file
+class DataCollectionViewModel @Inject internal constructor(
+    private val submissionRepository: SubmissionRepository,
+    private val locationOfInterestHelper: LocationOfInterestHelper
+) : AbstractViewModel() {
+    val submission: @Hot(replays = true) LiveData<Loadable<Submission>>
+    val jobName: @Hot(replays = true) LiveData<String>
+    val loiName: @Hot(replays = true) LiveData<String>
+
+    private val argsProcessor: @Hot(replays = true) FlowableProcessor<DataCollectionFragmentArgs> =
+        BehaviorProcessor.create()
+
+    init {
+        val submissionStream: Flowable<Loadable<Submission>> =
+            argsProcessor.switchMapSingle { args ->
+                submissionRepository.createSubmission(
+                    args.surveyId, args.locationOfInterestId, args.submissionId
+                ).map { Loadable.loaded(it) }.onErrorReturn { Loadable.error(it) }
```

Comment:
Instead of using `Loadable` here, can we just make the return value nullable and leverage nullability checks?

Reply:
Great idea! Done!